redfish-core/lib/license_service.hpp

```diff
@@ -0,0 +1,678 @@
+                            const bool* availablePtr =
+                                std::get_if<bool>(&propertyMap.second);
+                            if (availablePtr == nullptr)
+                            {
+                                messages::internalError(asyncResp->res);
+                                break;
+                            }
+                            available = *availablePtr;
+                        }
+                    }
+                }
+                if (interfaceMap.first == "xyz.openbmc_project.State.Decorator."
+                                          "OperationalStatus")
+                {
+                    for (const auto& propertyMap : interfaceMap.second)
+                    {
+                        if (propertyMap.first == "Functional")
+                        {
+                            const bool* functionalPtr =
+                                std::get_if<bool>(&propertyMap.second);
+                            if (functionalPtr == nullptr)
+                            {
+                                messages::internalError(asyncResp->res);
+                                break;
+                            }
+                            state = *functionalPtr;
+                        }
+                    }
+                }
+            }
+            asyncResp->res.jsonValue["@odata.type"] = "#License.v1_0_0.License";
+            asyncResp->res.jsonValue["@odata.id"] =
+                "/redfish/v1/LicenseService/Licenses/" + licenseEntryID;
+            asyncResp->res.jsonValue["Id"] = licenseEntryID;
+            asyncResp->res.jsonValue["SerialNumber"] = *serialNumPtr;
+            asyncResp->res.jsonValue["Name"] = *licenseNamePtr;
+            asyncResp->res.jsonValue["ExpirationDate"] =
+                redfish::time_utils::getDateTimeUint(expirationTime);
+            translateLicenseTypeDbusToRedfish(asyncResp, *licenseTypePtr);
+            translateAuthorizationTypeDbusToRedfish(asyncResp,
+                                                    *authorizationTypePtr);
+            asyncResp->res.jsonValue["MaxAuthorizedDevices"] = *deviceNumPtr;
+
+            if (available)
+            {
+                asyncResp->res.jsonValue["Status"]["Health"] = "OK";
+                if (state)
+                {
+                    asyncResp->res.jsonValue["Status"]["State"] = "Enabled";
+                }
+                else
+                {
+                    asyncResp->res.jsonValue["Status"]["State"] = "Disabled";
+                }
+            }
+            else
+            {
+                asyncResp->res.jsonValue["Status"]["Health"] = "Critical";
+                asyncResp->res.jsonValue["Status"]["State"] =
+                    "UnavailableOffline";
+            }
+        }
+    });
+}
+
+inline void requestRoutesLicenseEntry(App& app)
+{
+    BMCWEB_ROUTE(app, "/redfish/v1/LicenseService/Licenses/<str>/")
+        .privileges({{"Login"}})
```

Comment:
GetLicense here

Reply:
does it really add any value now. The name was same for all these time

Reply:
You are talking about function name right? If that is the case please tell me the convention we should follow. Quick check with recently approved commit show this name requestRoutesThermalMetrics(App& app)

Reply:
I am talking about the privileges

Reply:
If you want to rebase and use, I think it is better but I won't require it

Reply:
I am not planning to do this change in this PR. Will take it up as separate refactoring line item